### PR TITLE
docs: add github action deploy

### PR DESCRIPTION
**Pull request by kecrily**
close https://github.com/vuejs/vitepress/issues/789

### CODE REVIEW

File: docs/guide/deploying.md [hunk 1]
```diff
@@ -122,17 +124,52 @@ deploy:
...
+
+3. Now push commit to `master`, pages will deploy automaticlly
+
+## GitLab Pages and GitLab CI
 
-2. Set `outDir` in `.vitepress/config.js` to `../public`.
+1. Set `outDir` in `.vitepress/config.js` to `../public`.
 
-3. Create a file called `.gitlab-ci.yml` in the root of your project with the content below. This will build and deploy your site whenever you make changes to your content:
+2. Create a file called `.gitlab-ci.yml` in the root of your project with the content below. This will build and deploy your site whenever you make changes to your content:
```
Comment (xfq) [inline docs/guide/deploying.md:172]: You can probably keep all the numbers as `1.` and let the Markdown parser handle it automatically:

```
1. Set `outDir`...
1. Create a file called...
```

File: docs/guide/deploying.md [hunk 2]
```diff
@@ -15,6 +15,18 @@ The following guides are based on some shared assumptions:
 }
 ```
 
+
+::: tip Use pages hosting provided by Git service
+
+Services such as GitHub, GitLab and other git services provide page hosting. (GitHub is used as an example below)
```
Comment (xfq) [inline docs/guide/deploying.md:21]: Oxford comma should probably be added here (per [Vue Docs Writing Guide](https://github.com/vuejs/docs/blob/main/.github/contributing/writing-guide.md#grammar)).

File: docs/guide/deploying.md (hunk 1 above)
Comment (kecrily) [reply docs/guide/deploying.md:172]: I think it's important to keep markdown well readable without a renderer. So I choose to number them manually.

File: docs/guide/deploying.md (hunk 2 above)
Comment (kecrily) [reply docs/guide/deploying.md:21]: I just learned that the vue docs has such a writing guide, and I'm going to give it a good read.

File: docs/guide/deploying.md (hunk 1 above)
Comment (xfq) [reply docs/guide/deploying.md:172]: I don't think it makes much of a difference to readability, but I don't have a strong opinion here and will let the project's maintainers decide.

**Pull request closed without merging**